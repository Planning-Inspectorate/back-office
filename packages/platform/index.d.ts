import './config/dotenv';
import './http/ttl';
import './util/CryptoUtils';
import './util/date';
import './util/filter';
import './util/url';
import './validators/date';
import './validators/postcode';

declare module '@pins/platform' {
	export * from './config/dotenv';
	export * from './http/ttl';
	export * from './util/CryptoUtils';
	export * from './util/date';
	export * from './util/filter';
	export * from './util/url';
	export * from './validators/date';
	export * from './validators/postcode';
}
