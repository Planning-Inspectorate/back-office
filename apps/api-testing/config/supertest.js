// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
