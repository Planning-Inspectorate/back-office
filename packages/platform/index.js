export * from './http/hooks.js';
export * from './testing/assets/assets.js';
export * from './testing/fake.js';
export * from './testing/html-parser.js';
export * from './utils/date.js';
export * from './utils/dotenv.js';
export * from './utils/filter.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
