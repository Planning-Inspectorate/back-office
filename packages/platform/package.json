{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=16.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.0",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.1",
    "@types/lodash-es": "^4.17.6",
    "type-fest": "^2.12.2"
  }
}
