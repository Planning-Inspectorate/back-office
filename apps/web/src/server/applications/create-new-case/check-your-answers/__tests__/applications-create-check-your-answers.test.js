import { parseHtml } from '@pins/platform';
import nock from 'nock';
import supertest from 'supertest';
import { fixtureCases } from '../../../../../../testing/applications/fixtures/cases.js';
import { createTestEnvironment } from '../../../../../../testing/index.js';

const { app, installMockApi, teardown } = createTestEnvironment();
const request = supertest(app);
const errorPostResponse = {
	errors: {
		projectLocation: 'Missing projectLocation',
		subSector: 'Missing subSector',
		sector: 'Missing sector',
		regions: 'Missing regions',
		gridReferenceEasting: 'Missing gridReferenceEasting',
		gridReferenceNorthing: 'Missing gridReferenceNorthing'
	}
};

const nocks = () => {
	nock('http://test/').get('/applications/case-team').reply(200, {});
	nock('http://test/')
		.get(/\/applications\/1(.*)/g)
		.reply(200, fixtureCases[0]);

	nock('http://test/')
		.get(/\/applications\/2(.*)/g)
		.reply(200, fixtureCases[3]);
};

describe('applications create: check your answers', () => {
	beforeEach(installMockApi);
	afterEach(teardown);

	afterAll(() => {
		nock.cleanAll();
	});

	beforeEach(async () => {
		await request.get('/applications-service/case-team');
	});

	describe('Check your answers', () => {
		describe('GET /check-your-answers', () => {
			beforeEach(async () => {
				nocks();
			});

			const baseUrl = '/applications-service/create-new-case/1/check-your-answers';

			it('should render the page', async () => {
				const response = await request.get(baseUrl);
				const element = parseHtml(response.text);

				expect(element.innerHTML).toMatchSnapshot();
				expect(element.innerHTML).toContain('I accept - confirm creation of a new case');
			});
		});

		describe('POST /check-your-answers', () => {
			beforeEach(async () => {
				nocks();
			});

			const baseUrl = '/applications-service/create-new-case/1/check-your-answers';

			it('should confirm status if no fields are missing', async () => {
				nock('http://test/').post('/applications/1/start').reply(200, fixtureCases[0]);

				const response = await request.post(baseUrl);

				/* const element = parseHtml(response.text);
				expect(element.innerHTML).toMatchSnapshot(); */

				expect(response?.headers?.location).toContain(
					'/applications-service/create-new-case/1/case-created'
				);
			});

			it('should display errors if fields are missing', async () => {
				nock('http://test/').post('/applications/1/start').reply(200, errorPostResponse);

				const response = await request.post(baseUrl);
				const element = parseHtml(response.text);

				expect(element.innerHTML).toMatchSnapshot();
				expect(element.innerHTML).toContain('Enter the project location');
				expect(element.innerHTML).toContain('Choose the subsector of the project');
				expect(element.innerHTML).toContain('Choose the sector of the project');
				expect(element.innerHTML).toContain('Choose at least one region');
				expect(element.innerHTML).toContain('Enter the Grid reference Easting');
				expect(element.innerHTML).toContain('Enter the Grid reference Northing');
			});
		});
	});

	describe('Case created', () => {
		beforeEach(async () => {
			nocks();
		});

		const baseUrl = '/applications-service/create-new-case/2/case-created';

		it('should render the page', async () => {
			const response = await request.get(baseUrl);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
			expect(element.innerHTML).toContain('New case has been created');
		});
	});
});
