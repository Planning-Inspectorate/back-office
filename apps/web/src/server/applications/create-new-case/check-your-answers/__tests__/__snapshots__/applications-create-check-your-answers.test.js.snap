// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications create: check your answers Case created should render the page 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <div class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-two-thirds\\">
            <div class=\\"govuk-panel govuk-panel--confirmation\\">
                <h1 class=\\"govuk-panel__title\\"> New case has been created</h1>
                <div class=\\"govuk-panel__body\\">The case reference number
                    <br><strong>CASE/04</strong>
                </div>
            </div>
        </div>
    </div>
    <div>
        <br><a class=\\"govuk-link\\" href=\\"/applications-service/\\">Go back to main Applications homepage</a>
    </div>
</main>"
`;

exports[`applications create: check your answers Check your answers GET /check-your-answers should render the page 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <h1 class=\\"govuk-heading-l\\"><div class=\\"govuk-grid-row\\"><div class=\\"govuk-grid-column-full\\"> Check your answers before creating a new project</div></div></h1>
    <form method=\\"post\\" action=\\"\\">
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-grid-column-full\\">
                <table class=\\"govuk-table\\">
                    <thead class=\\"govuk-table__head\\">
                        <tr class=\\"govuk-table__row\\">
                            <th scope=\\"col\\" class=\\"govuk-table__header govuk-table__caption--m pins-table__cell--l govuk-!-static-padding-bottom-6\\"
                            colspan=\\"3\\">Project information</th>
                        </tr>
                    </thead>
                    <tbody class=\\"govuk-table__body\\">
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Project name</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">Title CASE/01</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a id=\\"name\\" class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Project description</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">Ipsum dolor sit amet consectetur adipiscing elit sed do eiusmod tempor
                                incididunt ut labore et dolore magna aliqua Ut enim ad minim veniam quis
                                nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat</td>
                            <td                             class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a id=\\"description\\" class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/\\">Change</a>
                                </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Sector</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">Transport EN</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a id=\\"sector\\" class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/sector\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Subsector</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">Highways EN</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a id=\\"subSector\\" class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/sub-sector\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Project location</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">London</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a id=\\"projectLocation\\" class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/geographical-information\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Grid reference Easting</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">123456</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a id=\\"gridReferenceEasting\\" class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/geographical-information\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Grid reference Northing</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">987654</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a id=\\"gridReferenceNorthing\\" class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/geographical-information\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Region(s)</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">London EN,Yorkshire EN</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a id=\\"regions\\" class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/regions\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Map zoom level</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">City EN</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/zoom-level\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Project email</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">some@ema.il</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/team-email\\">Change</a>
                            </td>
                        </tr>
                    </tbody>
                </table>
                <table class=\\"govuk-table\\">
                    <thead class=\\"govuk-table__head\\">
                        <tr class=\\"govuk-table__row\\">
                            <th scope=\\"col\\" class=\\"govuk-table__header govuk-table__caption--m pins-table__cell--l govuk-!-static-padding-bottom-6\\"
                            colspan=\\"3\\">Applicant information</th>
                        </tr>
                    </thead>
                    <tbody class=\\"govuk-table__body\\">
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Organisation name</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">Org name</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/applicant-organisation-name\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Contact name</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">John Smith</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/applicant-full-name\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Address</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">Applicant address, London, ABC123</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/applicant-address\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Website</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">website.web</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/applicant-website\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Email address</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">email@email.co</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/applicant-email\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Telephone number</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">001</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/applicant-telephone-number\\">Change</a>
                            </td>
                        </tr>
                    </tbody>
                </table>
                <table class=\\"govuk-table\\">
                    <thead class=\\"govuk-table__head\\">
                        <tr class=\\"govuk-table__row\\">
                            <th scope=\\"col\\" class=\\"govuk-table__header govuk-table__caption--m pins-table__cell--l govuk-!-static-padding-bottom-6\\"
                            colspan=\\"3\\">Key dates</th>
                        </tr>
                    </thead>
                    <tbody class=\\"govuk-table__body\\">
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Anticipated submission date published</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\"></td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/key-dates\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Anticipated submission date internal</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\"></td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/key-dates\\">Change</a>
                            </td>
                        </tr>
                    </tbody>
                </table>
            </div>
        </div>
        <br>
        <div class=\\"govuk-warning-text\\"><span class=\\"govuk-warning-text__icon\\" aria-hidden=\\"true\\">!</span>
            <strong             class=\\"govuk-warning-text__text\\"><span class=\\"govuk-warning-text__assistive\\">Warning</span> Creating the
                project will not publish the page onto the NSIP website</strong>
        </div>
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\">
                <button class=\\"govuk-button\\" data-module=\\"govuk-button\\">I accept - confirm creation of a new case</button>
            </div>
        </div>
    </form>
</main>"
`;

exports[`applications create: check your answers Check your answers POST /check-your-answers should display errors if fields are missing 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <div class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-two-thirds\\">
            <div class=\\"govuk-error-summary\\" data-module=\\"govuk-error-summary\\">
                <div role=\\"alert\\">
                    <h2 class=\\"govuk-error-summary__title\\"> There is a problem</h2>
                    <div class=\\"govuk-error-summary__body\\">
                        <ul class=\\"govuk-list govuk-error-summary__list\\">
                            <li><a href=\\"#projectLocation\\">Enter the project location</a>
                            </li>
                            <li><a href=\\"#subSector\\">Choose the subsector of the project</a>
                            </li>
                            <li><a href=\\"#sector\\">Choose the sector of the project</a>
                            </li>
                            <li><a href=\\"#regions\\">Choose at least one region</a>
                            </li>
                            <li><a href=\\"#gridReferenceEasting\\">Enter the Grid reference Easting</a>
                            </li>
                            <li><a href=\\"#gridReferenceNorthing\\">Enter the Grid reference Northing</a>
                            </li>
                        </ul>
                    </div>
                </div>
            </div>
        </div>
    </div>
    <h1 class=\\"govuk-heading-l\\"><div class=\\"govuk-grid-row\\"><div class=\\"govuk-grid-column-full\\"> Check your answers before creating a new project</div></div></h1>
    <form method=\\"post\\" action=\\"\\">
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-grid-column-full\\">
                <table class=\\"govuk-table\\">
                    <thead class=\\"govuk-table__head\\">
                        <tr class=\\"govuk-table__row\\">
                            <th scope=\\"col\\" class=\\"govuk-table__header govuk-table__caption--m pins-table__cell--l govuk-!-static-padding-bottom-6\\"
                            colspan=\\"3\\">Project information</th>
                        </tr>
                    </thead>
                    <tbody class=\\"govuk-table__body\\">
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Project name</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">Title CASE/01</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a id=\\"name\\" class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Project description</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">Ipsum dolor sit amet consectetur adipiscing elit sed do eiusmod tempor
                                incididunt ut labore et dolore magna aliqua Ut enim ad minim veniam quis
                                nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat</td>
                            <td                             class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a id=\\"description\\" class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/\\">Change</a>
                                </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Sector</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">Transport EN</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a id=\\"sector\\" class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/sector\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Subsector</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">Highways EN</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a id=\\"subSector\\" class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/sub-sector\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Project location</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">London</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a id=\\"projectLocation\\" class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/geographical-information\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Grid reference Easting</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">123456</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a id=\\"gridReferenceEasting\\" class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/geographical-information\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Grid reference Northing</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">987654</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a id=\\"gridReferenceNorthing\\" class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/geographical-information\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Region(s)</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">London EN,Yorkshire EN</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a id=\\"regions\\" class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/regions\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Map zoom level</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">City EN</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/zoom-level\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Project email</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">some@ema.il</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/team-email\\">Change</a>
                            </td>
                        </tr>
                    </tbody>
                </table>
                <table class=\\"govuk-table\\">
                    <thead class=\\"govuk-table__head\\">
                        <tr class=\\"govuk-table__row\\">
                            <th scope=\\"col\\" class=\\"govuk-table__header govuk-table__caption--m pins-table__cell--l govuk-!-static-padding-bottom-6\\"
                            colspan=\\"3\\">Applicant information</th>
                        </tr>
                    </thead>
                    <tbody class=\\"govuk-table__body\\">
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Organisation name</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">Org name</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/applicant-organisation-name\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Contact name</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">John Smith</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/applicant-full-name\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Address</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">Applicant address, London, ABC123</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/applicant-address\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Website</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">website.web</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/applicant-website\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Email address</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">email@email.co</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/applicant-email\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Telephone number</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\">001</td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/applicant-telephone-number\\">Change</a>
                            </td>
                        </tr>
                    </tbody>
                </table>
                <table class=\\"govuk-table\\">
                    <thead class=\\"govuk-table__head\\">
                        <tr class=\\"govuk-table__row\\">
                            <th scope=\\"col\\" class=\\"govuk-table__header govuk-table__caption--m pins-table__cell--l govuk-!-static-padding-bottom-6\\"
                            colspan=\\"3\\">Key dates</th>
                        </tr>
                    </thead>
                    <tbody class=\\"govuk-table__body\\">
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Anticipated submission date published</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\"></td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/key-dates\\">Change</a>
                            </td>
                        </tr>
                        <tr class=\\"govuk-table__row\\">
                            <td class=\\"govuk-table__cell font-weight--700 pins-table__cell--s\\">Anticipated submission date internal</td>
                            <td class=\\"govuk-table__cell pins-table__cell--s\\"></td>
                            <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/1/key-dates\\">Change</a>
                            </td>
                        </tr>
                    </tbody>
                </table>
            </div>
        </div>
        <br>
        <div class=\\"govuk-warning-text\\"><span class=\\"govuk-warning-text__icon\\" aria-hidden=\\"true\\">!</span>
            <strong             class=\\"govuk-warning-text__text\\"><span class=\\"govuk-warning-text__assistive\\">Warning</span> Creating the
                project will not publish the page onto the NSIP website</strong>
        </div>
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\">
                <button class=\\"govuk-button\\" data-module=\\"govuk-button\\">I accept - confirm creation of a new case</button>
            </div>
        </div>
    </form>
</main>"
`;
